### PR TITLE
Fixes for using node version 4.

### PR DESCRIPTION
**Pull request by rolandpoulter**

### CODE REVIEW

File: lib/common/encryption.js [hunk 1]
```diff
@@ -78,11 +78,31 @@ function encryptionFactory(
...
+     * ['sha512', 'sha256','md5'], the default is 'sha512'
+     * @param {String} [salt] - The salt feed to hash algorithm, if not specified, the function will
+     * generate a random salt.
+     * @return {String} the salt
+     */
+    Encryption.prototype.createSalt = function (algorithm, salt) {
         algorithm = algorithm || 'sha512';
-        salt = salt || crypt3.createSalt(algorithm);
-        return crypt3(data, salt);
+        var signitures = {
```
Comment (benbp) [inline lib/common/encryption.js]: `signatures` is the correct spelling

File: lib/common/encryption.js [hunk 2]
```diff
@@ -78,11 +78,31 @@ function encryptionFactory(
...
+          md5: '$1$',
```
Comment (benbp) [inline lib/common/encryption.js]: 4 space indentation plz :P

**Comment by benbp** [comment]
Are there unit tests that need updating as well?

**Comment by benbp** [comment]
Thanks for fixing!

**Comment by rolandpoulter** [comment]
@benbp don't think I need to add more unit tests. I could for `createSalt` but `createHash` already has tests. I will if requested.

**Comment by benbp** [comment]
Oh I guess I meant did we need to update existing ones? If not, great. :+1: when the build is passing.

**Comment by jfrey** [comment]
+1

**Comment by yyscamper** [comment]
Even it has been merged, I still want to raise a :+1: , glad we have move to Node 4 ,thank for your wok @rolandpoulter

**Pull request merged**